nuc_morph_analysis/lib/preprocessing/add_features.py

```diff
@@ -140,6 +140,63 @@ def add_feature_at(df, frame_column, feature, feature_column, multiplier=1):
         )
     return df
 
+def add_mean_feature_over_trajectory(df, feature_list, multiplier_list):
+    """
+    Add the mean of a given feature over the growth trajectory 
+    from transition to frame breakdown. 
+    
+    Parameters
+    ----------
+    df : DataFrame
+        The dataframe
+    feature_list : list
+        List of column names
+    multiplier_list : list
+        List of scale to multiply the mean by
+        
+    Returns
+    -------
+    df : DataFrame
+        The dataframe with the added mean feature columns
+    """
+    for feature, multiplier in zip(feature_list, multiplier_list):
+        for tid, dft in df.groupby("track_id"):
+            start = dft.frame_transition.unique()[0]
```

Comment:
`dft.frame_transition.unique()[0]` seems needlessly slow. why not just run `dft.frame_transition.values[0]`?
same comment applies to the line below. `stop = dft.Fb.unique()[0]`. 

I'm not too concerned about speed but anything that runs through every single track will take some time.

Reply:
Done! I know there are other ways this could be sped up but when I tried doing apply / lamda method I wasnt getting the result that I expected so I went with the simplest for loop method.